htdocs/fourn/class/fournisseur.facture-rec.class.php

```diff
@@ -0,0 +1,2209 @@
+                if ($status == self::STATUS_SUSPENDED) {
+                    $statusType = 'status6';
+                    $labelStatus = $langs->transnoentitiesnoconv('Disabled');
+                } else {
+                    $statusType = 'status4';
+                    $labelStatus = $langs->transnoentitiesnoconv('Active');
+                }
+            } else {
+                if ($status == self::STATUS_SUSPENDED) {
+                    $statusType = 'status6';
+                    $labelStatus = $langs->transnoentitiesnoconv('Disabled');
+                } else {
+                    $statusType = 'status0';
+                    $labelStatus = $langs->transnoentitiesnoconv('Draft');
+                }
+            }
+        } elseif ($mode == 4) {
+            $prefix = '';
+            if ($recur) {
+                if ($status == self::STATUS_SUSPENDED) {
+                    $statusType = 'status6';
+                    $labelStatus = $langs->transnoentitiesnoconv('Disabled');
+                } else {
+                    $statusType = 'status4';
+                    $labelStatus = $langs->transnoentitiesnoconv('Active');
+                }
+            } else {
+                if ($status == self::STATUS_SUSPENDED) {
+                    $statusType = 'status6';
+                    $labelStatus = $langs->transnoentitiesnoconv('Disabled');
+                } else {
+                    $statusType = 'status0';
+                    $labelStatus = $langs->transnoentitiesnoconv('Draft');
+                }
+            }
+        } elseif ($mode == 5 || $mode == 6) {
+            $prefix = '';
+            if ($mode == 5) {
+                $prefix = 'Short';
+            }
+            if ($recur) {
+                if ($status == self::STATUS_SUSPENDED) {
+                    $statusType = 'status6';
+                    $labelStatus = $langs->transnoentitiesnoconv('Disabled');
+                } else {
+                    $statusType = 'status4';
+                    $labelStatus = $langs->transnoentitiesnoconv('Active');
+                }
+            } else {
+                if ($status == self::STATUS_SUSPENDED) {
+                    $statusType = 'status6';
+                    $labelStatus = $langs->transnoentitiesnoconv('Disabled');
+                } else {
+                    $statusType = 'status0';
+                    $labelStatus = $langs->transnoentitiesnoconv('Draft');
+                }
+            }
+        }
+
+        $labelStatusShort = $labelStatus;
+
+        return dolGetStatus($labelStatus, $labelStatusShort, '', $statusType, $mode);
+    }
+
+    /**
+     *  Initialise an instance with random values.
+     *  Used to build previews or test instances.
+     *	id must be 0 if object instance is a specimen.
+     *
+     *	@param	string		$option		''=Create a specimen invoice with lines, 'nolines'=No lines
+     *  @return	void
+     */
+    public function initAsSpecimen($option = '')
```

Comment:
Je crois que cette fonction n'a pas été adaptée pour les factures modèles fournisseur (je ne sais pas si on en a besoin ici ou pas)  car on trouve plus loin des `new FactureLigne($this->db);`

Reply:
Je ne crois pas qu'on en ai besoin, elle n'est pas utilisé pour l'objet FactureFournisseurRec
Je supprime sa déclaration également ?

Reply:
Done